src/main/java/duke/command/DeadlineCommand.java

```diff
@@ -0,0 +1,41 @@
+package duke.command;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.DukeException;
+
+/** A subclass of Command which adds a deadline to the taskList */
+public class DeadlineCommand extends AddCommand {
+    private final String name;
+    private final String schedule;
+
+    /**
+     * Creates a DeadlineCommand.
+     *
+     * @param content The task information supplied by the user.
+     * @throws DukeException If the content has missing name or schedule or the format does not contain /by.
+     */
+    public DeadlineCommand(String content) throws DukeException {
+        if (!content.contains("/by")) {
+            throw new DukeException("☹ OOPS!!! The information contains invalid delimiter");
+        } else {
+            String[] contentParts = content.split(" /by ");
+            if (contentParts[0].equals("") || contentParts[1].equals("")) {
```

Comment:
Perhaps giving a comment on the expected format of the input (so that contentParts[0] and contentParts[1] is understandable to reviewers)